java/code/src/com/redhat/rhn/manager/setup/ProductSyncManager.java

```diff
@@ -98,70 +104,131 @@
      * Adds multiple products.
      * @param productIdents the product ident list
      * @param user the current user
-     * @throws ProductSyncException if an error occurred
      * @return a map of added products and an error message if any
      */
-    public Map<String, Optional<? extends Throwable>> addProducts(List<String> productIdents, User user) {
+    public Map<String, Optional<? extends Exception>> addProducts(List<String> productIdents, User user) {
+        // Extract the ids from the identifiers
+        final List<Long> productsId = productIdents.stream().map(
+            // Ident are in the form 'product_id-label'. If the product id is negative, the ident starts with a minus,
+            // so we need to skip the first one when performing the indexOf
+            id -> Long.valueOf(id.substring(0, id.indexOf('-', id.startsWith("-") ? 1 : 0)))
```

Comment:
Would this have the same result?

```suggestion
            id -> Long.valueOf(id.substring(0, id.indexOf('-', 1)))
```
 The first might be '-' or not. If we always ignore the first character it should also work for product ids between 0 and 9.

Reply:
Yeah, indeed you are right. I will simplify this.